server/schedule/operator_controller.go

```diff
@@ -472,3 +507,71 @@ func (o *OperatorRecords) Put(op *Operator, status pdpb.OperatorStatus) {
 	}
 	o.ttl.Put(id, record)
 }
+
+// exceedStoreLimit returns true if the store exceeds the cost limit after adding the operator. Otherwise, returns false.
+func (oc *OperatorController) exceedStoreLimit(ops ...*Operator) bool {
+	opInfluence := NewTotalOpInfluence(ops, oc.cluster)
+	for storeID := range opInfluence.storesInfluence {
+		if oc.storesLimit[storeID] == nil {
+			rate := oc.cluster.GetStoreBalanceRate()
+			oc.newStoreLimit(storeID, rate)
+		}
+		stepCost := opInfluence.GetStoreInfluence(storeID).StepCost
+		if stepCost == 0 {
+			continue
+		}
+		available := oc.storesLimit[storeID].Available()
+		storeLimit.WithLabelValues(strconv.FormatUint(storeID, 10), "available").Set(float64(available) / float64(RegionInfluence))
+		if available < stepCost {
+			oc.cluster.SetStoreOverload(storeID)
+			return true
+		}
+	}
+	return false
+}
+
+// SetAllStoresLimit is used to set limit of all stores.
+func (oc *OperatorController) SetAllStoresLimit(rate float64) {
+	oc.Lock()
+	defer oc.Unlock()
+	for storeID := range oc.storesLimit {
+		oc.newStoreLimit(storeID, rate)
+	}
+}
+
+// SetStoreLimit is used to set the limit of a store.
+func (oc *OperatorController) SetStoreLimit(storeID uint64, rate float64) {
+	oc.Lock()
+	defer oc.Unlock()
+	oc.newStoreLimit(storeID, rate)
+}
+
+// newStoreLimit is used to create the limit of a store.
+func (oc *OperatorController) newStoreLimit(storeID uint64, rate float64) {
+	var capacity int64
+	rate = rate * float64(RegionInfluence)
+	if rate > float64(RegionInfluence) {
+		capacity = int64(rate)
+	} else {
+		capacity = RegionInfluence
+	}
+	oc.storesLimit[storeID] = ratelimit.NewBucketWithRate(rate, capacity)
+}
+
+// GetAllStoresLimit is used to get limit of all stores.
+func (oc *OperatorController) GetAllStoresLimit() map[uint64]interface{} {
```

Comment:
I think better to return `map[uint64]float64` and convert to json in package api.